src/main/java/seedu/command/attendance/AddAttendanceList.java

```diff
@@ -39,77 +45,101 @@ public void addToList() throws PacException {
+            withFlag();
+        } else {
+            lineByLine();
+        }
+    }
+
+    /**
+     * Method to create new attendanceList using flags within the user input.
+     * @throws PacException If parameter provided is invalid.
+     */
+    private void withFlag() throws PacException {
+        while (!input.toLowerCase().equals("done")) {
+            UI.display("Please key following format:\n"
                     + "n/Name p/Status[Y/N]");
-                ui.readUserInput();
-                input = ui.getUserInput();
-                if (input.equals("done")) {
-                    break;
-                }
-                attendances.addToList(attendanceParser.parseAttendance(input), eventName);
-                newStudentList.addToList(name);
-                studentNumber++;
+            ui.readUserInput();
+            input = ui.getUserInput();
+            if (input.equals("done")) {
+                break;
             }
-        } else {
-            while (!status.equals("done")) {
-                UI.display("Please key in student name.");
-                ui.readUserInput();
-                name = ui.getUserInput();
-                if (name.equals("done")) {
-                    break;
-                }
-                if (attendances.isDuplicate(name)) {
-                    UI.display("Duplicated name found, student name : [ " + name + " ] not added");
-                } else {
-                    UI.display("To mark the student as present, please use 'y' or 'Y'.");
-                    UI.display("By default the student will be marked as absent,"
+            attendanceList.addToList(attendanceParser.parseAttendance(input), eventName);
+            studentName = attendanceParser.parseAttendance(input).getStudentName();
+            newStudentList.addToList(studentName);
+            studentNumber++;
+        }
+        studentListCollection.push(newStudentList);
+        UI.display("You have successfully added "
+                + studentNumber + " to the attendance list.\n");
+    }
+
+    /**
+     * Method to create new attendanceList using a line-by-line input from the user.
+     * @throws PacException If parameter provided is invalid.
+     */
+    private void lineByLine() throws PacException {
+        while (!studentName.equals("done")) {
+            UI.display("Please key in student name.");
+            ui.readUserInput();
+            studentName = ui.getUserInput();
+            if (studentName.equals("done")) {
+                break;
+            }
+            if (attendanceList.isDuplicate(studentName)) {
+                throw new PacException("Duplicated name found, student name : [ " + studentName + " ] not added");
+            } else {
+                UI.display("To mark the student as present, please use 'y' or 'Y'.");
+                UI.display("By default the student will be marked as absent,"
                         + " if any other input is given.");
```

Comment:
Might want to add a method in UI to display this two sentences. then use ui...() to look neater.

Reply:
Made the changes to combine